### ISSUE TITLE
ENH: Better error message when interpolating dtype integer with method="linear"

### ISSUE THREAD

**Issue by AartGoossens** [pandas-dev/pandas#41565]
#### Is your feature request related to a problem?

When trying to interpolate a column with dtype integer with `method="linear"` the error message is not very descriptive (example to reproduce at the bottom of this issue):
```
ValueError: Invalid fill method. Expecting pad (ffill) or backfill (bfill). Got linear
```

As "linear" is the default interpolation method I think the error message should be more descriptive, even though IntegerArray is still experimental according to the [docs](https://pandas.pydata.org/pandas-docs/stable/user_guide/integer_na.html).

#### Describe the solution you'd like

I think the error message should include a hint to the potential cause that the array you're trying to interpolate is of dtype integer. Something like this could work:
```
ValueError: Invalid fill method. Expecting pad (ffill) or backfill (bfill). Got linear. Are you trying to interpolate an integer column?
```
I would be happy to create a PR with the improvement myself.

#### API breaking implications

As this only requires a change to the error message it does not break any API.

#### Describe alternatives you've considered

An alternative solution would be to implement linear interpolation for integer arrays but I suspect that that is not compatible/unexpected with some use-cases of integer arrays.

#### Additional context
Reproducable example:
```python
import pandas as pd


s = pd.Series([1, None, 3], dtype=pd.Int64Dtype())

s.interpolate(method="linear")
```

**Comment by mzeitlin11** [comment]
Thanks for the clear report @AartGoossens! There are currently some issues with interpolation with the new nullable data types (#40252) - like you say implementing interpolation should likely be the end goal, but for now a more explicit error message would be an improvement.

**Comment by AartGoossens** [comment]
@mzeitlin11 Does the "Contributions Welcome" milestone mean that it's OK if I would create a PR with the suggested change?

**Comment by mzeitlin11** [comment]
Yep! (though even without a milestone contributions are always welcome :)

**Comment by MichaelTiemannOSC** [comment]
I just got this message from an attempt to use linear interpolation with ExtensionArrays.  I don't think the error message is the problem, rather the wholesale lack of anything behind the implementation of linear interpolation in version 1.4.0 for ExtensionArrays.  linear interpolation works great for floats (despite the gratuitous error message) but for ExtensionArrays, the code just calls to fillna, which doesn't have a linear method.  That's a big gap!

**Comment by tamargrey** [comment]
Also running into this error when the `category` dtype is used. I expect this may be the time when this error is expected, as a linear interpolation of categorical data doesn't make sense to me

**Comment by jbrockmendel** [comment]
This works on main.  could use a test (first check to see if one already exists, which I expect it does)

**Comment by ShigrafS** [comment]
@jbrockmendel  Has the test already been implemented?

**Comment by jbrockmendel** [comment]
Not that I'm aware of.

**Comment by jiangmuran** [comment]
I would like to work on this by improving the error message for the `linear` fill method to hint about integer interpolation. I will use automated tools to assist and will disclose their use in the PR per the contribution policy.
